unit-tests/src/test/scala/java/lang/ClassSuite.scala

```diff
@@ -135,4 +135,22 @@ object ClassSuite extends tests.Suite {
     assert(!classOf[java.lang.Class[_]].isInterface)
     assert(classOf[java.lang.Runnable].isInterface)
   }
+
+  private def assertEqualClass(l: java.lang.Class[_],
+                               r: java.lang.Class[_]): Unit = {
+    assert(l == r, s"$l != $r")
+    assert(System.identityHashCode(l) == System.identityHashCode(r),
+           s"$l diff ident hash $r")
```

Comment:
What you really want here is reference equality (`eq`). The `identityHashCode` is not necessary:
```suggestion
    assert(l eq r, s"$l ne $r")
```